src/Diagnostics.DataProviders/DataProviders/Base/SupportObserverDataProviderBase.cs

```diff
@@ -63,16 +63,38 @@ public Task<dynamic> GetResource(string resourceUrl)
                 throw new FormatException(exceptionMessage, ex);
             }
 
-            if (uri.Host.Contains(allowedHosts[0]) || uri.Host.Contains(allowedHosts[1]) || Configuration.ObserverLocalHostEnabled)
+            if (uri.Host.Contains(allowedHosts[0]) || uri.Host.Contains(allowedHosts[1]))
             {
                 return GetWawsObserverResourceAsync(uri);
             }
+            else if (Configuration.ObserverLocalHostEnabled && TryGetLocalObserverRoute(uri, out var newUri))
+            {
+                return GetWawsObserverResourceAsync(newUri);
+            }
             else
             {
                 return GetSupportObserverResourceAsync(uri);
             }
         }
 
+        private static bool TryGetLocalObserverRoute(Uri uri, out Uri localObserverUri)
+        {
+            localObserverUri = uri;
+
+            if (!uri.AbsolutePath.StartsWith("/observer"))
+            {
+                return true;
+            }
+
+            if (uri.AbsolutePath.Contains("subscriptions") && (uri.AbsolutePath.EndsWith("webspaces") || uri.AbsolutePath.EndsWith("sites")))
+            {
+                localObserverUri = new Uri(uri, uri.AbsolutePath.Remove(0, "/observer".Length));
```

Comment:
If I am following the logic correctly, in the case that the Uri is something like `"https://support-bay-api.azurewebsites.net/observer/subscriptions/.../webspaces/..."` then this will make the localObserverUri equal the absolute path of the support-bay-api URL absent the `/observer` string. Once this request object make it eventually to `SendObserverRequestAsync` then the HttpClient base URI to the LocalHostObserver will be overided with this Uri's absolute path and the HttpClient will send the request to the SupportBayApi.

Am I understanding this correctly? Is this the intentional behavior?

Reply:
Yes, exactly

Reply:
Can we verify what APIs are already there in the SelfHostObserver? I did some work sometime back where I migrated a good chunk of APIs that existing detectors use from support-bay-api into SelfHostObserver